pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/ServerCnx.java

```diff
@@ -171,24 +182,33 @@ protected void handlePartitionMetadataRequest(CommandPartitionedTopicMetadata pa
         }
         final long requestId = partitionMetadata.getRequestId();
         final String topic = partitionMetadata.getTopic();
-        getPartitionedTopicMetadata(getBrokerService().pulsar(), getRole(), DestinationName.get(topic))
-                .thenAccept(metadata -> {
-                    int partitions = metadata.partitions;
-                    ctx.writeAndFlush(Commands.newPartitionMetadataResponse(partitions, requestId));
-                }).exceptionally(ex -> {
-                    if (ex instanceof PulsarClientException) {
-                        log.warn("Failed to authorize {} at [{}] on topic {} : {}", getRole(), remoteAddress, topic,
-                                ex.getMessage());
-                        ctx.writeAndFlush(Commands.newPartitionMetadataResponse(ServerError.AuthorizationError,
-                                ex.getMessage(), requestId));
-                    } else {
-                        log.warn("Failed to get Partitioned Metadata [{}] {}: {}", remoteAddress, topic,
-                                ex.getMessage(), ex);
-                        ctx.writeAndFlush(Commands.newPartitionMetadataResponse(ServerError.ServiceNotReady,
-                                ex.getMessage(), requestId));
-                    }
-                    return null;
-                });
+        if (service.getLookupRequestSemaphore().tryAcquire()) {
```

Comment:
To avoid indenting the whole section, what about doing: 

```java
if (!service.getLookupRequestSemaphore().tryAcquire()) {
    /// deal with error
    return;
}

// continue normal
```